keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1475,225 +1382,400 @@ prevented. Deleting the entire cluster also leaves resources allocated outside
+    // allows users to define a preference, without making it a hard requirement.
+    //
+    //
+    // Score *SomeScoringStruct
+}
+
+// Request is a request for one of many resources required for a claim.
+// This is typically a request for a single resource like a device, but can
+// also ask for several identical devices. It might get extended to support
+// asking for one of several different alternatives.
+type Request struct {
+    // The name can be used to reference this request in a pod.spec.containers[].resources.claims
+    // entry and in a constraint of the claim.
+    //
+    // Must be a DNS label.
+    Name string
+
+    *RequestDetail
+
+    // FUTURE EXTENSION:
+    //
+    // OneOf contains a list of requests, only one of which must be satisfied.
+    // Requests are listed in order of priority.
     //
-    // The object must be in the same namespace as the ResourceClaim.
     // +optional
-    ParametersRef *ResourceClaimParametersReference
+    // +listType=atomic
+    // OneOf []RequestDetail
 }
-```
 
-The `ResourceClassName` field may be left empty. The parameters are sufficient
-to determine which driver needs to provide resources. This leads to some corner cases:
-- Empty `ResourceClassName` and nil `ParametersRef`: this is a claim which requests
-  no resources. Such a claim can always be allocated with an empty result. Allowing
-  this simplifies code generators which dynamically fill in the resource requests
-  because they are allowed to generate an empty claim.
-- Non-empty `ResourceClassName`, nil `ParametersRef`, nil
-  `ResourceClass.DefaultClaimParametersRef`: this is handled the same way, the
-  only difference is that the cluster admin has decided that such claims need
-  no resources by not providing default parameters.
-
-There is no default ResourceClass. If that is desirable, then it can be
-implemented with a mutating and/or admission webhook.
+type RequestDetail struct {
+    // When referencing a DeviceClass, a request inherits additional
+    // configuration and selectors.
+    //
+    // +optional
+    DeviceClassName *string
 
-```
-// ResourceClaimStatus tracks whether the resource has been allocated and what
-// the resulting attributes are.
-type ResourceClaimStatus struct {
-    // Allocation is set by the resource driver once a resource or set of
-    // resources has been allocated successfully. If this is not specified, the
-    // resources have not been allocated yet.
+    // Each selector must be satisfied by a device which is requested.
+    //
     // +optional
-    Allocation *AllocationResult
+    // +listType=atomic
+    Selectors []Selector
 
-    // ReservedFor indicates which entities are currently allowed to use
-    // the claim. A Pod which references a ResourceClaim which is not
-    // reserved for that Pod will not be started.
+    *Amount // inline, so no extra level in YAML and no separate documentation
```

Comment:
Why not just put those fields here?

Reply:
Just habits, I'll inline.

Reply:
Actually, there is a reason. It is a *pointer* on purpose, not some embedded struct, see https://github.com/kubernetes/enhancements/pull/4709#discussion_r1640552271.

Reply:
responded there